Composer/packages/client/src/recoilModel/dispatchers/application.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { CallbackInterface, useRecoilCallback } from 'recoil';
+import debounce from 'lodash/debounce';
+
+import { appUpdateState, announcementState, onboardingState, creationFlowStatusState } from '../atoms/appState';
+import { AppUpdaterStatus, CreationFlowStatus } from '../../constants';
+import OnboardingState from '../../utils/onboardingStorage';
+import { StateError, AppUpdateState } from '../../recoilModel/types';
+
+import { setError } from './shared';
+
+export const applicationDispatcher = () => {
+  const setAppUpdateStatus = useRecoilCallback(
+    ({ set, snapshot: { getPromise } }: CallbackInterface) => async (
+      status: AppUpdaterStatus,
+      version: string | undefined
+    ) => {
+      const currentAppUpdate = await getPromise(appUpdateState);
+      const newAppUpdateState = {
+        ...currentAppUpdate,
+      };
+      if (status === AppUpdaterStatus.UPDATE_AVAILABLE) {
+        newAppUpdateState.version = version;
+      }
+      if (status === AppUpdaterStatus.IDLE) {
+        newAppUpdateState.progressPercent = 0;
+        newAppUpdateState.version = undefined;
+      }
+
+      newAppUpdateState.status = status;
+
+      set(appUpdateState, {
+        ...newAppUpdateState,
+        status,
```

Comment:
I don't think this line is necessary because `status` is already assigned to `newAppUpdateState.status` on line 33.

This is the equivalent of:

```
set(appUpdateState, {
   ...{ status },
   status,
});
```